openverse_catalog/dags/providers/factory_utils.py

```diff
@@ -138,3 +140,34 @@ def pull_media_wrapper(
     duration = end_time - start_time
     ti.xcom_push(key="duration", value=duration)
     return data
+
+
+def date_partition_for_prefix(
+    schedule_interval: str | None, logical_date: datetime
+) -> str:
+    """
+    Given a schedule interval and the logical date for a DAG run, determine an
+    appropriate partition for the run. This partition will be used as part of the S3
+    key prefix for a given TSV.
+
+    Prefix mapping (schedule interval to partition):
+        - Hourly -> `year=YYYY/month=MM/day=DD`
+        - Daily -> `year=YYYY/month=MM`
+        - None/yearly/monthly/weekly/other -> `year=YYYY`
+    """
+    # Always partition by year
+    prefix = f"year={logical_date.year}"
```

Comment:
Total nit, but I would move this line below the schedule string/cron variables, so that it's with the month/day prefixes. The way it is structured currently makes it a little easy to miss.